ext/RMagick/rmimage.c

```diff
@@ -7113,29 +7113,33 @@ Image_get_pixels(VALUE self, VALUE x_arg, VALUE y_arg, VALUE cols_arg, VALUE row
                  , columns, rows, x, y);
     }
 
-    // Cast AcquireImagePixels to get rid of the const qualifier. We're not going
-    // to change the pixels but I don't want to make "pixels" const.
-    exception = AcquireExceptionInfo();
-    pixels = GetVirtualPixels(image, x, y, columns, rows, exception);
-    CHECK_EXCEPTION()
-
-    (void) DestroyExceptionInfo(exception);
-
     // If the function failed, return a 0-length array.
-    if (!pixels)
+    if (columns * rows <= 0)
     {
         return rb_ary_new();
     }
 
     // Allocate an array big enough to contain the PixelPackets.
-    size = (long)(columns * rows);
-    pixel_ary = rb_ary_new2(size);
+    pixel_ary = rb_ary_new2(columns * rows);
```

Comment:
Hmm, I think `pixel_ary` just requires `(columns - x) * (rows - y)` pixel size. `columns * rows` is too much.

Reply:
I'm guessing the original `for (n = 0; n < size; n++)` is also wrong...
if `x` or `y` is not 0, `pixel_ary` will has unexpected value in trailing region....